contracts/test/PlonkVerifier.t.sol

```diff
@@ -0,0 +1,432 @@
+// SPDX-License-Identifier: Unlicensed
+
+/* solhint-disable contract-name-camelcase, func-name-mixedcase, one-contract-per-file */
+/* solhint-disable no-inline-assembly  */
+
+// NOTE: For developers and auditors: we mainly test the consistency between the outputs between
+// Solidity and Jellyfish library, with the help of fuzzer-generated inputs from Forge Testing.
+// Inside the logic of `batchVerify()`, variables values only need to be consistent and valid
+// (i.e. valid group or field elements) and don't need to be from a correct proof/public input.
+// Only the last step `_batchVerifyOpeningProof` will test *correctness* of these parameters.
+// Therefore, we employ more randomly generated dummy inputs for most tests for robustness,
+// and only relie on Rust-code to generate correct inputs for the `_batchVerifyOpeningProof`.
+
+pragma solidity ^0.8.0;
+
+// Libraries
+import "forge-std/Test.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { IPlonkVerifier } from "../src/interfaces/IPlonkVerifier.sol";
+import { VkTest } from "./stubs/Transfer1In2Out24DepthVk.sol";
+import { PolynomialEval as Poly } from "../src/libraries/PolynomialEval.sol";
+
+// Target contract
+import { PlonkVerifier as V } from "../src/libraries/PlonkVerifier.sol";
+
+/// @dev Common helpers/utils for PlonkVerifier tests
+contract PlonkVerifierCommonTest is Test {
+    /// @dev Sanitize a single value to be valid scalar field Bn254::Fr.
+    function sanitizeScalarField(uint256 a) public view returns (uint256) {
+        a = bound(a, 0, BN254.R_MOD - 1);
+        BN254.validateScalarField(a);
+        return a;
+    }
+
+    /// @dev Sanitize all values in `a` to be valid scalar fields Bn254::Fr.
+    /// This is helpful to sanitize fuzzer-generated random `uint[]` values.
+    function sanitizeScalarFields(uint256[] memory a) public view returns (uint256[] memory) {
+        for (uint256 i = 0; i < a.length; i++) {
+            a[i] = bound(a[i], 0, BN254.R_MOD - 1);
```

Comment:
Wonder why this doesn't call the function above.

Reply:
sure, I will use `sanitizeScalarField(a[i])` instead